tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchReindexingTests.java

```diff
@@ -0,0 +1,263 @@
+        expectResultsFromQuery(queryString, dataUser.getAdminUser());
+
+        // WHEN
+        // Run reindexer against the initial documents.
+        reindex(Map.of("ALFRESCO_REINDEX_JOB_NAME", "reindexByIds",
+                       "ELASTICSEARCH_INDEX_NAME", CUSTOM_ALFRESCO_INDEX,
+                       "ALFRESCO_REINDEX_FROM_ID", "0",
+                       "ALFRESCO_REINDEX_TO_ID", "1000"));
+
+        // THEN
+        // Check system document is indexed.
+        Utility.sleep(1000, 10000, () -> {
+            expectResultsFromQuery(queryString, dataUser.getAdminUser(), "budget.xls");
+        });
+        LOGGER.info("End test");
+    }
+
+    @Test(groups = TestGroup.SEARCH)
+    public void testReindexerFixesBrokenIndex() throws Exception
+    {
+        // GIVEN
+
+        // Delete all documents inside Elasticsearch.
+        cleanUpIndex();
+        //stop live indexing
+        AlfrescoStackInitializer.liveIndexer.stop();
+        // Create document.
+
+        String testStart = DateTimeFormatter.ofPattern("yyyyMMddHHmm").format(ZonedDateTime.now(Clock.systemUTC()));
+        String documentName = createDocument();
+        // Check document not indexed.
+        // Nb. The cm:name:* term ensures that the query hits the index rather than the db.
+
+        String queryString = "=cm:name:" + documentName + " AND cm:name:*";
+        Utility.sleep(1000, 10000, () -> {
+            expectResultsFromQuery(queryString, dataUser.getAdminUser());
+        });
+
+        // WHEN
+        // Run reindexer (leaving ALFRESCO_REINDEX_TO_TIME as default).
+        reindex(Map.of("ALFRESCO_REINDEX_JOB_NAME", "reindexByDate",
+                       "ELASTICSEARCH_INDEX_NAME", CUSTOM_ALFRESCO_INDEX,
+                       "ALFRESCO_REINDEX_FROM_TIME", testStart));
+
+        // THEN
+        // Check document indexed.
+        Utility.sleep(1000, 10000, () -> {
+            expectResultsFromQuery(queryString, dataUser.getAdminUser(), documentName);
+        });
+
+        // TIDY
+        // Restart ElasticsearchConnector.
+        AlfrescoStackInitializer.liveIndexer.start();
+
+    }
+
+    @Test(groups = TestGroup.SEARCH)
+    public void testRecreateIndex() throws Exception
+    {
+        // GIVEN
+        // Create document.
+        String documentName = createDocument();
+        // Stop ElasticsearchConnector.
+        AlfrescoStackInitializer.liveIndexer.stop();
+        // Delete index documents.
+        cleanUpIndex();
+
+        // WHEN
+        // Run reindexer (with default dates to reindex everything).
+        reindex(Map.of("ALFRESCO_REINDEX_JOB_NAME", "reindexByDate",
+                       "ELASTICSEARCH_INDEX_NAME", CUSTOM_ALFRESCO_INDEX));
+
+        // THEN
+        // Check document indexed.
+        // Nb. The cm:name:* term ensures that the query hits the index rather than the db.
+        Utility.sleep(1000, 10000, () -> {
```

Comment:
Because we are using Utility.sleep almost always with expectResultsFromQuery can we move that statement in the method?